test/jdk/java/nio/channels/spi/SelectorProvider/inheritedChannel/UnixDomainChannelTest.java

```diff
@@ -21,41 +21,50 @@
  * questions.
  */
 
+import java.io.IOException;
+import java.net.SocketAddress;
+import java.net.UnixDomainSocketAddress;
 import java.nio.channels.*;
 import java.nio.ByteBuffer;
-import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import static java.net.StandardProtocolFamily.UNIX;
 import static java.nio.charset.StandardCharsets.ISO_8859_1;
 
 /*
- * Make sure that System.inheritedChannel returns null when given a UNIX domain socket
+ * Make sure that System.inheritedChannel returns the correct type
  */
 
 public class UnixDomainChannelTest {
+    private static final UnixDomainSocketAddress SOCK_ADDR =
+            UnixDomainSocketAddress.of(Path.of("foo.socket"));
+
+    private static boolean passed = true;
 
     public static class Child {
         public static void main(String[] args) throws Exception {
-            // we just want to make sure that System.inheritedChannel either
-            // returns a connected channel, or null if it is given a listener
+            // we want to make sure that System.inheritedChannel either
+            // returns a ServerSocketChannel or a SocketChannel
             Channel channel = System.inheritedChannel();
             String result = channel == null ? "N" : "Y";
             if (args[0].equals("test1") || args[0].equals("test2")) {
-                // socket is writeable
-                ByteChannel bc = (ByteChannel)channel;
-                ByteBuffer buf = ByteBuffer.wrap(result.getBytes(ISO_8859_1));
-                bc.write(buf);
+                if (channel instanceof SocketChannel) {
+                    SocketChannel sc = (SocketChannel) channel;
+                    ByteBuffer buf = ByteBuffer.wrap(result.getBytes(ISO_8859_1));
+                    sc.write(buf);
+                }
```

Comment:
Should the test fail if `channel` is not an instance of `SocketChannel`? Or at least print some kind of debug information?
Do we want to verify that it's a Unix Domain Socket Channel - e.g. by looking at the local address?
Same question for the server socket case below...